### PR TITLE
zabbix check v2 first version

### PR DESCRIPTION
**Pull request by vishalxebia**

### CODE REVIEW

File: zabbix/README.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+   Agent's configuration directory to start collecting your zabbix performance data.
+   See the sample [zabbix.d/conf.yaml][2] for all available configuration options.
+
+2. Restart the Agent
+
+## Data Collected
+
+### Metrics
+
+Zabbix does not include any metrics.
```
Comment (TizianoPerrucci) [inline zabbix/README.md]: update Readme to specify what data gets collected

File: zabbix/stackstate_checks/zabbix/data/conf.yaml.example [hunk 2]
```diff
@@ -0,0 +1,9 @@
+init_config:
+    min_collection_interval: 60 # run every minute
```
Comment (TizianoPerrucci) [inline zabbix/stackstate_checks/zabbix/data/conf.yaml.example]: in agent v2 `min_collection_interval` has been move to the `instance`

File: zabbix/stackstate_checks/zabbix/zabbix.py [hunk 3]
```diff
@@ -0,0 +1,362 @@
...
+                'tags': [
+                    'host_id:%s' % host_id,
+                    'host:%s' % zabbix_host.host,
+                    'host_name:%s' % zabbix_host.name,
+                    'severity:%s' % severity,
+                    'triggers:%s' % triggers
+                ]
+            })
+
+        self.stop_snapshot()
```
Comment (TizianoPerrucci) [inline zabbix/stackstate_checks/zabbix/zabbix.py]: even if there are errors we should make sure `stop_snapshot` is called
have a look at other checks how we do that

File: zabbix/tests/test_zabbix.py [hunk 4]
```diff
@@ -0,0 +1,494 @@
...
+        component = topo_instances['components'][0]
+        self.assertEqual(component['data']['domain'], 'Zabbix')
+        labels = component['data']['labels']
+        for label in ['zabbix', 'host group:Zabbix servers', 'host group:MyHostGroup']:
+            if label not in labels:
+                self.fail("Component does not have label '%s'." % label)
+
+    def test_zabbix_problems(self):
+
+        # TODO this is needed because the topology retains data across tests
```
Comment (TizianoPerrucci) [inline zabbix/tests/test_zabbix.py]: misleading comment

File: zabbix/tests/test_zabbix.py [hunk 5]
```diff
@@ -0,0 +1,494 @@
...
+            if tag not in tags:
+                self.fail("Event does not have tag '%s', got: %s." % (tag, tags))
+        self.assertEqual(len(tags), 5)
+
+    def test_zabbix_no_problems(self):
+        """
+        When there are no problems, we are expecting all host components to go to green.
+        To make this happen we need to send an event that says all is OK.
+        """
+        # TODO this is needed because the topology retains data across tests
```
Comment (TizianoPerrucci) [inline zabbix/tests/test_zabbix.py]: misleading comment

File: zabbix/tests/test_zabbix.py [hunk 6]
```diff
@@ -0,0 +1,494 @@
...
+            if tag not in tags:
+                self.fail("Event does not have tag '%s', got: %s." % (tag, tags))
+        self.assertEqual(len(tags), 5)
+
+    def test_zabbix_determine_most_severe_state(self):
+        """
+            A host can have multiple active problems.
+            From the active problems we determine the most severe state and send that to StackState
+        """
+        # TODO this is needed because the topology retains data across tests
```
Comment (TizianoPerrucci) [inline zabbix/tests/test_zabbix.py]: misleading comment

File: zabbix/stackstate_checks/zabbix/zabbix.py (hunk 3 above)
Comment (vishalxebia) [reply zabbix/stackstate_checks/zabbix/zabbix.py]: made it to always call this.

File: zabbix/tests/test_zabbix.py (hunk 4 above)
Comment (vishalxebia) [reply zabbix/tests/test_zabbix.py]: fixed.

File: zabbix/tests/test_zabbix.py (hunk 6 above)
Comment (vishalxebia) [reply zabbix/tests/test_zabbix.py]: fixed.

File: zabbix/tests/test_zabbix.py (hunk 5 above)
Comment (vishalxebia) [reply zabbix/tests/test_zabbix.py]: fixed.

File: zabbix/stackstate_checks/zabbix/data/conf.yaml.example (hunk 2 above)
Comment (vishalxebia) [reply zabbix/stackstate_checks/zabbix/data/conf.yaml.example]: removed.

File: zabbix/README.md (hunk 1 above)
Comment (vishalxebia) [reply zabbix/README.md]: done.

File: zabbix/stackstate_checks/zabbix/zabbix.py [hunk 7]
```diff
@@ -0,0 +1,366 @@
...
+        self.ssl_verify = instance.get('ssl_verify', True)
+
+        url = instance['url']
+
+        topology_instance = {
+            "type": self.SERVICE_CHECK_NAME,
+            "url": url
+        }
+        try:
+            self.check_connection(url)
```
Comment (TizianoPerrucci) [inline zabbix/stackstate_checks/zabbix/zabbix.py]: can the `check_connection` or `login` fail?
if it does the `stop_snapshot` is called without a `start_snapshot` first, that may cause problems
also can we test this ?

Comment (vishalxebia) [reply zabbix/stackstate_checks/zabbix/zabbix.py]: moved `start_snapshot` to first line in `try` block to avoid the failure.

**Pull request merged**